### PR TITLE
windows: removed deprecated features in 2.6

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Removed features deprecated in older versions and set for removal in 2.6

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_chocolatey
win_feature
win_iis_webapppool
win_package
win_regedit

##### ANSIBLE VERSION
```
2.6
```

### CODE REVIEW

File: changelogs/fragments/windows-deprecation-removal.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+deprecated_features:
+- win_chocolatey - removed deprecated upgrade option and choco_* output return values
```
Comment (gundalow) [inline changelogs/fragments/windows-deprecation-removal.yaml]: As this gets converted into markdown I think you may need

```
- `win_chocolatey` - removed...
```

**Review by gundalow — COMMENTED** [review]
Could you please add an entry into the porting_guides, thanks m

File: lib/ansible/modules/windows/win_iis_webapppool.ps1 [hunk 2]
```diff
@@ -1,24 +1,11 @@
...
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-# WANT_JSON
-# POWERSHELL_COMMON
+# Copyright: (c) 2015, Jon Hawkesworth (@jhawkesworth) <figs@unity.demon.co.uk>
```
Comment (jhawkesworth) [inline lib/ansible/modules/windows/win_iis_webapppool.ps1]: seem to have lost attribution for Henrik - I merely did some fixing, but Henrik is original module author:

-# (c) 2015, Henrik Wallström <henrik@wallstroms.nu>

**Comment by jhawkesworth** [comment]
This looks good to me, I'm in favour of merging now so this can be tested during 2.6 development.

**Comment by gundalow** [comment]
+1 to removals & deprecation notices being done as early in the release cycle as possible.

File: changelogs/fragments/windows-deprecation-removal.yaml (hunk 1 above)
Comment (jborean93) [reply changelogs/fragments/windows-deprecation-removal.yaml]: Hmm, will see what @nitzmahone says, I haven't been doing them for my previous fragment entries but can start doing it now. Will also add to the porting guide as well.

File: lib/ansible/modules/windows/win_iis_webapppool.ps1 (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_iis_webapppool.ps1]: My bad, will add it back in.

File: changelogs/fragments/windows-deprecation-removal.yaml (hunk 1 above)
Comment (gundalow) [reply changelogs/fragments/windows-deprecation-removal.yaml]: oh, it's actually generated as RST
https://raw.githubusercontent.com/ansible/ansible/stable-2.5/changelogs/CHANGELOG-v2.5.rst
I can see `win\_audit\_rule` as well as

```
The `stat`` and ``win_stat`` modules have changed
```

So maybe ignore the backticks for the moment, and we can check with @nitzmahone later?

File: changelogs/fragments/windows-deprecation-removal.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+deprecated_features:
```
Comment (nitzmahone) [inline changelogs/fragments/windows-deprecation-removal.yaml]: should be `removed_features`

File: changelogs/fragments/windows-deprecation-removal.yaml (hunk 1 above)
Comment (nitzmahone) [reply changelogs/fragments/windows-deprecation-removal.yaml]: I haven't been too careful about formatting; for module bugfixes, I've been doing just `module_name - description of thing` like what's already here... We're still kinda playing around with what kinds of post-processing we want to do on these too, so this is fine with me for now. We can always go back and reformat consistently later if we come up with some standards.

**Review by nitzmahone — COMMENTED** [review]
Changes look fine (other than the one thing in the changelog).

I had some ideas on how to implement deprecation warnings on return values, but it definitely wouldn't be 2.6 material (would require some surgery in the result processing and Jinja magic that's orthogonal to "stabilization" ;) ). Question is: if that were coming in 2.7, would we want to wait until 2.8 to nuke any/all of these retvals? I think the `restart_required`->`reboot_required` is probably the most likely one to actually be used in a playbook for something. So maybe we should hold off on that one until 2.8, and if I can get the retval dep warnings working in 2.7, at least people will have one release worth of runtime warnings. Thoughts?

**Comment by jborean93** [comment]
That sounds fine to me, happy to keep the return values until a later date, in reality we did change the deprecation timeframe from 2 to 4 releases after we deprecated these so that keeps it in line.

File: changelogs/fragments/windows-deprecation-removal.yaml (hunk 3 above)
Comment (jborean93) [reply changelogs/fragments/windows-deprecation-removal.yaml]: Thanks, wasn't sure if there was a key but now I know.

**Comment by nitzmahone** [comment]
You want to just update whichever ones we're keeping then to say "2.8" then? Hopefully I can have retval deprecation working for 2.7...

**Comment by jborean93** [comment]
Sure, will update it later today along with the other issues.

File: changelogs/fragments/windows-deprecation-removal.yaml (hunk 1 above)
Comment (jborean93) [reply changelogs/fragments/windows-deprecation-removal.yaml]: Have kept the formatting the same but added entries to the porting guide.

File: lib/ansible/modules/windows/win_iis_webapppool.ps1 (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_iis_webapppool.ps1]: Thanks for the pickup, should be fixed now.

**Review by nitzmahone — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Review by dharmabumstead — APPROVED** [review]
Approved with a minor bit of surgery.  Thanks @jborean93!

**Comment by jborean93** [comment]
Thanks everyone for the review. The test failure is unrelated to the Windows changes so will merge through.

**Pull request merged**